### PR TITLE
Enable Versioning checks to decide between running all versions or just the latest

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
Aims to speed up Versioning checks on PRs by allowing only the latest version to be tested against, with all versions saved for overnight builds.

### CODE REVIEW

File: .ci/code/Versioning_Test/Verify/FromJson.cs
Comment (FraserGreenroyd) [inline .ci/code/Versioning_Test/Verify/FromJson.cs:49]:
```suggestion
```

File: .ci/code/Versioning_Test/Verify/FromJson.cs
Comment (IsakNaslundBh) [inline .ci/code/Versioning_Test/Verify/FromJson.cs:47]: Wonder if this should not be defaulting to true, to keep the current behaviour, unless instructed otherwise by the caller. Will still approve and leave it up to you if you want to change this or not.

```suggestion
        public static TestResult FromJsonDatasets(bool testAll = true)
```

**Review by IsakNaslundBh — APPROVED** [review]
Approved, good to get the speed up. See slight comment on potential change though.

Leave it up to you if you want to change or not.

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: core, installer, versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: ready-to-merge

**Pull request merged**
